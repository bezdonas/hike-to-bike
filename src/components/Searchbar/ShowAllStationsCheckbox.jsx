import { h, Component } from 'preact';
import './show-all.css';

// ShowAllStationsCheckboxProps = {
//  showAllStations: boolean,
//  toggleShowAllStations: function,
// }
export default class ShowAllStationsCheckbox extends Component {
  render(props) {
    const { showAllStations, toggleShowAllStations } = this.props;
    return (
      <div class="show-all-wrapper">
        <label>
          Show all stations{' '}
          <input
            type="checkbox"
            onClick={toggleShowAllStations}
            value={showAllStations}
          />
        </label>
      </div>
    );
  }
}
