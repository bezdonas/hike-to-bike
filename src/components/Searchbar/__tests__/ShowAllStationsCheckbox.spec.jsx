import { h, preact } from 'preact';
import 'preact-compat';
import render from 'preact-render-to-json';
import ShowAllStationsCheckbox from '../ShowAllStationsCheckbox.jsx';

describe('ShowAllStationsCheckbox', () => {
  it('renders as expected and matches snapshot', () => {
    const props = {
      showAllStations: true,
      toggleShowAllStations: () => false,
    };
    expect(render(<ShowAllStationsCheckbox {...props} />)).toMatchSnapshot();
  });
});
