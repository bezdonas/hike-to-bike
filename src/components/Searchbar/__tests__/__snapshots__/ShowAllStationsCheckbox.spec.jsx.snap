// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ShowAllStationsCheckbox renders as expected and matches snapshot 1`] = `
<div
  class="show-all-wrapper"
>
  <label>
    Show all stations 
    <input
      onClick={[Function]}
      type="checkbox"
      value={true}
    />
  </label>
</div>
`;
